src/main/clojure/clara/rules/engine.cljc

```diff
@@ -72,14 +74,14 @@
 ;; Returns a new session with the additional facts inserted.
 (defprotocol ISession
 
-  ;; Inserts a fact.
-  (insert [session fact])
+  ;; Inserts facts.
+  (insert [session facts])
 
-  ;; Retracts a fact.
+  ;; Retracts facts.
   (retract [session fact])
 
   ;; Fires pending rules and returns a new session where they are in a fired state.
-  (fire-rules [session])
+  (fire-rules [session opts])
```

Comment:
I did that in the fire-rules function in rules.cljc.  I've always thought of that as the public API for this.

````
rules.cljc
(defn fire-rules
  "in real version the doc is here."
  ([session] (eng/fire-rules session {}))
  ([session opts] (eng/fire-rules session opts)))
````